api/orb/client.go

```diff
@@ -0,0 +1,142 @@
+package orb
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/CircleCI-Public/circleci-cli/api"
+	"github.com/CircleCI-Public/circleci-cli/api/graphql"
+	"github.com/CircleCI-Public/circleci-cli/settings"
+	"github.com/pkg/errors"
+)
+
+var (
+	once        sync.Once
+	client      Client
+	clientError error
+)
+
+type clientVersion string
+
+// ConfigResponse is a structure that matches the result of the GQL
+// query, so that we can use mapstructure to convert from
+// nested maps to a strongly typed struct.
+type QueryResponse struct {
+	OrbConfig struct {
+		api.ConfigResponse
+	}
+}
+
+type Client interface {
+	OrbQuery(configPath string, ownerId string) (*api.ConfigResponse, error)
+}
+
+func GetClient(config *settings.Config) (Client, error) {
+	once.Do(func() {
+		client, clientError = newClient(config)
+	})
+
+	return client, clientError
+}
```

Comment:
As we discussed offline, I think this not-quite-singleton pattern is actually more harmful than having potentially more than one client.

The risk here is a caller calling `GetClient` a second time, with a different config, and getting a `Client` not suited to their config, without error.

In this implementation, I think it's worth having callers rely on an explicit `NewClient(config)` even if it means wastefully creating clients, rather than using `GetClient` expecting certain results and facing hard-to-undertand bugs.